usr/share/rear/lib/layout-functions.sh

```diff
@@ -611,3 +611,207 @@ retry_command ()
+    fi
+
+    if [ -z "$MIGRATION_MODE" ] ; then
+        return 0
+    fi
+
+    # We temporarily map all devices in the mapping to new names _REAR[0-9]+_
+    replaced_count=0
+    replacement_file="$TMP_DIR/replacement_file"
+    : > "$replacement_file"
+
+    # Generate replacements.
+    while read source target junk ; do
+        if ! has_replacement "$source" ; then
+            add_replacement "$source"
+        fi
+
+        if ! has_replacement "$target" ; then
+            add_replacement "$target"
+        fi
+    done < "$MAPPING_FILE"
+
+    # Replace all originals with their replacements.
+    while read original replacement junk ; do
+        # Replace partitions (we normalize cciss/c0d0p1 to _REAR5_1)
+        part_base="$original"
+        case "$original" in
+            *mmcblk[0-9]*|*nvme[0-9]*n[1-9]*|*rd[/!]c[0-9]*d[0-9]*|*cciss[/!]c[0-9]*d[0-9]*|*ida[/!]c[0-9]*d[0-9]*|*amiraid[/!]ar[0-9]*|*emd[/!][0-9]*|*ataraid[/!]d[0-9]*|*carmel[/!][0-9]*)
+                part_base="${original}p" # append p between main device and partitions
+                ;;
+            *mapper[/!]*)
+                case $OS_VENDOR in
+                    SUSE_LINUX)
+                    # SUSE Linux SLE12 put a "-part" between [mpath device name] and [part number].
+                    # For example /dev/mapper/3600507680c82004cf8000000000000d8-part1.
+                    # But SLES11 uses a "_part" instead. (Let's assume it is the same for SLES10 )
+                    if (( $OS_VERSION < 12 )) ; then
+                        # For SUSE before version 12
+                        part_base="${original}_part" # append _part between main device and partitions
+                    else
+                        # For SUSE 12 or above
+                        part_base="${original}-part" # append -part between main device and partitions
+                    fi
+                    ;;
+                    RedHatEnterpriseServer)
+                        # RHEL 7 and above seems to named partitions on multipathed devices with
+                        # [mpath device name] + [part number] like standard disk.
+                        # For example: /dev/mapper/mpatha1
+
+                        # But the scheme in RHEL 6 need a "p" between [mpath device name] and [part number].
+                        # For exemple: /dev/mapper/mpathap1
+                        if (( $OS_VERSION < 7 )) ; then
+                            part_base="${original}p" # append p between main device and partitions
+                        fi
+                    ;;
+                    Ubuntu)
+                        # Ubuntu 16.04 (need to check for other version) named muiltipathed partitions with
+                        # [mapth device name] + "-part" + [part number]
+                        # for example : /dev/mapper/mpatha-part1
+                        part_base="${original}-part" # append -part between main device and partitions
+                    ;;
+
```

Comment:
What about Arch, Gentoo etc.?

Reply:
I don't have Arch or Gentoo (not available on POWER), so I don't know.